enumerable_methods.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    while self[i]
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    my_each { |e| yield(e, index(e)) }
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    selected_items = []
+    my_each { |item| selected_items << item if yield(item) }
+    selected_items
+  end
+
+  def my_all?(*args)
+    return grep(args.first).length == size unless args.empty?
+
+    my_each { |i| return false unless yield(i) } if block_given?
+    my_each { |i| return false unless i } unless block_given?
+    true
+  end
+
+  def my_any?(*args)
+    return !grep(args.first).empty? unless args.empty?
+
+    my_each { |i| return true if yield(i) } if block_given?
+
+    my_each { |i| return true if i } unless block_given?
+
+    false
+  end
```

Comment:
In `my_any?`
- my_any? when no block or argument is given returns true if at least one of the collection is not false or nil